client/test/backend.go

```diff
@@ -32,6 +36,7 @@ type (
 		mu           sync.Mutex
 		latestEvents map[channel.ID]channel.AdjudicatorEvent
 		eventSubs    map[channel.ID][]chan channel.AdjudicatorEvent
+		balances     map[addressMapKey]map[assetMapKey]*big.Int
```

Comment:
Why did you use `map[addr]map[asset]` instead of `map[addrAssetTuple]`?  
This would simplify the logic in `add/get/setBalance`.

Reply:
It's semantically clearer for me to handle it in two maps.